### PR TITLE
More file close related fixes

### PR DESCRIPTION
**Pull request by rodrigc**
Moved from https://github.com/twisted/twisted/pull/306

https://twistedmatrix.com/trac/ticket/8573

Split into three commits for easier review: first adds some closing where it wasn't previously explicitly done at all, second changes more closing to use "with", and third is self explanatory.

### CODE REVIEW

**Comment by rodrigc** [comment]
@scop @markrwilliams I was able to get some of your changes committed in https://github.com/twisted/twisted/commit/56e1c4a559639643f9e9366612984da0b5f3f68c .  However, I could not get all your fixes committed.

The Twisted maintainers have implemented some backend checks for code coverage and will block any commits which do not have 100% of the patch covered by code.  This is painful because some of the existing code in Twisted does not have tests, but the Twisted maintainers are trying to increase the number of tests covering the code.  For example if you look here:

https://codecov.io/gh/twisted/twisted/compare/56e1c4a559639643f9e9366612984da0b5f3f68c...fc10a05a4f9e34e8bfdb3481ca86f36450286251

You will see that only 24% of the patch is covered.

Do you think you would be able to modify some tests or add new tests so that 100% of your patch is covered?

You can look at the coverage in your tree by doing the following.
Make sure you have tox and coverage installed:

```
pip install coverage tox
```

Go to the root of your twisted checkout and do:

```
coverage erase
tox -e py27-coverage
coverage combine
ls -l .coverage
coverage report
coverage html
```

You can then look at the coverage report in HTML format and see the parts
of your patch that need additional tests for coverage.

**Comment by rodrigc** [comment]
@scop @markrwilliams updated coverage report is here: https://codecov.io/gh/twisted/twisted/compare/0cf15cbd53ddab65388f21f3d60e1e86765545f2...628977b4c536aeab098a3cdddeaed833920fe75a

**Comment by scop** [comment]
I may look into extending the tests sometime, but doing that is much more work than I'm willing to commit right now.

I managed to get coverage to run using your instructions (+ doing the route changes detailed in https://twistedmatrix.com/trac/ticket/7494#comment:4 and commenting out the last test_loopback failer), but I'm not sure how to usefully read the report in order to pinpoint the changes that need additional test cases. Is there a way I can generate a comparison report like the one you linked to for my in-progress changes?

**Comment by rodrigc** [comment]
@scop I'm not sure how to generate a comparison report.  The comparison reports are generated by the https://codecov.io web site which has multiple coverage reports from the same GitHub project, so can generate a comparison report.

For a standalone user, the best bet is to look at the HTML generated report.  For lines that are green, that means that they have test coverage.  Lines that are pink/red do not have test coverage.

I have only been active in Twisted development for the past month, and have seen the Twisted devs start pushing more aggressively for 100% test coverage in new patches.  I can understand your perspective, where you are trying to fix something in existing code which does not have test coverage, and then the requirement to write a new test increases the work.

Do your best, based on your available time and interest, and keep the patches coming in.

File: twisted/application/app.py [hunk 1]
```diff
@@ -309,12 +309,18 @@ def runReactorWithLogging(config, oldstdout, oldstderr, profiler=None,
         else:
             reactor.run()
     except:
+        close = False
```
Comment (adiroiban) [inline twisted/application/app.py:312]: minor comment: can we have here `file = None`

File: twisted/mail/alias.py [hunk 2]
```diff
@@ -742,7 +742,8 @@ def __init__(self, items, *args):
                 except:
                     log.err("Invalid filename in alias file %r" % (addr[1:],))
                 else:
-                    addr = ' '.join([l.strip() for l in f])
+                    with f:
```
Comment (adiroiban) [inline twisted/mail/alias.py:745]: minor comment: maybe we can get rid of the else branch and just use `with open(addr[1:]) as f:` ... I don't see why we need the else branch here.

Comment (adiroiban) [inline twisted/mail/alias.py:745]: well... we need it to not catch join errors ... and strip() errors... so is ok to have it this way. sorry for the noise

**Comment by rodrigc** [comment]
I tried to merge this after review approval from @markrwilliams and was blocked by codecov

**Comment by rodrigc** [comment]
OK, now codecov let me through, and I was able to merge!

**Pull request merged**